OpenSim/Common/Test/testComponentInterface.cpp

```diff
@@ -1047,6 +1048,103 @@ void testComponentPathNames()
     top.connect();
 }
 
+void testTraversePathToComponent() {
+    class Base : public Component {
+        OpenSim_DECLARE_ABSTRACT_OBJECT(Base, Component);
+    public:
+        // Allow use of this protected function for testing purposes:
+        using Component::traversePathToComponent;
```

Comment:
Not sure why this would be necessary, since `Component::getComponent<C>` delegates to `traversePathToComponent()` and the not found cases throw a specify exception that you could `ASSERT_THROW` instead of testing for `nullptr`. That seems a bit cleaner to me.

Reply:
Done.